bin/publish

```diff
@@ -1,51 +1,70 @@
 #!/bin/bash
 
+if [[ -z "${TAG_NAME:-}" ]]; then
+  echo "Please supply environment variable TAG_NAME."
+  echo "If you see this error in Jenkins it means the publish script was run"
+  echo "for a build that wasn't triggered by a tag - please check publish stage conditions."
+  exit 1
+fi
+
 set -e
```

Comment:
Is there a reason that `-e` is below the usage? can't hurt to have that run there as well.